tests/load_tests/locustfile.py

```diff
@@ -0,0 +1,45 @@
+import os
+from locust import HttpUser, task, between
+from dotenv import load_dotenv
+
+
+class TestUser(HttpUser):
+    wait_time = between(0.5, 2)
+
+    def credentials(self):
+        load_dotenv()
+        test_username = os.getenv("AUTH_USERNAME")
+        test_password = os.getenv("AUTH_PASSWORD")
+        return test_username, test_password
+
+    @task
+    def health(self):
+        self.client.get("/health")
+
+    @task
+    def authenticate(self):
+        test_username, test_password = self.credentials()
+        self.client.post("/auth/token", data={"username": test_username, "password": test_password})
+
+    @task
+    def get_conversation(self):
+        test_username, test_password = self.credentials()
+        response = self.client.post(
+            "/auth/token", data={"username": test_username, "password": test_password}
+        )
+        token = response.json()["access_token"]
+        response = self.client.get("/conversation", headers={"Authorization": f"Bearer {token}"})
+
+    @task
+    def ask_question(self):
+        test_username, test_password = self.credentials()
+        response = self.client.post(
```

Comment:
Won't calling the auth endpoint distort the test? We need to know the time of question-answering.

Reply:
Locust logs the time of each request, not each test, so the test should not be distorted. So we will get two separate times. I've made a page in notion to record the results from one test, as an example of how the results look (I only ran the ask_question test): https://www.notion.so/LoadTesting-e32ea69c73a3491dafd9c4384899b579